packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,74 @@
+import { cn } from "@/lib/utils";
+import { useState } from "react";
+import * as React from "react";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+interface ConnectButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  buttonText?: string;
+  icon?: JSX.Element;
+  handleClick: () => void;
+  onSuccess?: () => void;
+  onError: () => void;
+}
+
+export function ConnectButton({
+  buttonText,
+  handleClick,
+  onSuccess,
+  onError,
+  className,
+  icon,
+}: ConnectButtonProps & AtomsGlobalConfigProps) {
+  const [isProcessing, setIsProcessing] = useState<boolean>(false);
+  const [isSuccess, setIsSuccess] = useState<boolean>(false);
+  const [errMsg, setErrMsg] = useState<string>("");
+
+  function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
```

Comment:
look at this again, I'm not sure we need handleSubmit at all here, not sure how we plan to use this component but for now we should just use onClick passed with button html props already

Reply:
and then handle different variants of the button when the user passes props like disabled, loading, errMsg, children...